src/lcm/argmax.py

```diff
@@ -113,13 +146,13 @@ def segment_argmax(a, segment_ids, num_segments):
 
     # Check where the array attains its maximum
     # ==================================================================================
-    max_value_mask = a == segment_max_expanded
+    max_value_mask = data == segment_max_expanded
 
     # Create index array of argmax indices for each segment (has same shape as a)
```

Comment:
```suggestion
    # Create index array of argmax indices for each segment (has same shape as data)
```

Are we sure that a unique value is returned in case of ties for maxima? *(assuming we rely on that implicitly in interpolations etc)* Do we have a test for that?

Reply:
I've added a test for `argmax` and `segment_argmax` and a corresponding note in the code.